src/request.rs

```diff
@@ -41,33 +40,31 @@ impl<'a> Request<'a> {
     /// Returns a `RequestResult` indicating whether the request was sent and processed successfully
     /// (according to the rules of XML-RPC).
     pub fn call(&self, client: &Client, url: &str) -> RequestResult {
-        self.call_with(client, url, |b| b)
+        self.call_with(client, url, |_| {})
     }
 
     /// Calls the method, giving the closure a chance to amend the hyper
     /// `RequestBuilder` before sending.
     pub fn call_with<F>(&self, client: &Client, url: &str, cb: F) -> RequestResult
-        where F: Fn(RequestBuilder) -> RequestBuilder
+        where F: Fn(&mut RequestBuilder) -> ()
```

Comment:
The old signature could be preserved if really needed, but as ownership of the builder has to be given back anyway and as far as I know everything relevant can be done with a mut reference this variant is probably more ergonomic to use.

Reply:
I'm fine with this change, but the `-> ()` can be removed.